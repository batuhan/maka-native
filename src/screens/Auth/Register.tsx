import React from "react";
import { View, Text } from "react-native";
import { connect } from "react-redux";

import useForm from "../../lib/components/useForm";
import validate from "../../lib/validations/registerForm";

import { fetchRegisterRequest } from "../../modules/auth/actions";
import strings from "../../config/strings";

import Button from "../../components/Button/Button";
import FormInput from "../../components/FormInput/FormInput";

const RegisterScreen = ({ register }: { register: () => void }) => {
  const { values, errors, handleChange, handleSubmit } = useForm(
    register,
    validate
  );

  return (
    <View>
      <FormInput
        name="email"
        value={values.email}
        error={errors.email}
        onChangeText={handleChange}
        placeholder={strings.EMAIL_PLACEHOLDER}
      />
      <FormInput
        name="fullname"
        value={values.fullname}
        error={errors.fullname}
        onChangeText={handleChange}
        placeholder={strings.NAME_PLACEHOLDER}
      />
      <FormInput
        name="password"
        value={values.password}
        error={values.password}
        onChangeText={handleChange}
        placeholder={strings.PASSWORD_PLACEHOLDER}
      />
      <Button onPress={handleSubmit}>
        <Text>{strings.REGISTER}</Text>
      </Button>
    </View>
  );
};

const mapStateToProps = (response: any) => {
  return response.authReducer;
};

const mapDispatchToProps = (dispatch: any) => {
  return {
    register: (email: string, password: string, fullname: string) => {
      dispatch(fetchRegisterRequest({ email, password, fullname }));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RegisterScreen);
