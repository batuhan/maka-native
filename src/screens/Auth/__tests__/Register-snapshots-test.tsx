import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";

import Register from "../Register";

const initialState = {};
const mockStore: any = configureStore();

describe("Register", () => {
  it("renders correctly", () => {
    const register = jest.fn();
    const instance = renderer.create(
      <Provider store={mockStore(initialState)}>
        <Register register={register} />{" "}
      </Provider>
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });
});
