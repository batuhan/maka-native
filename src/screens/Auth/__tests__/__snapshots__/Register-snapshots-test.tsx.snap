// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders correctly 1`] = `
Array [
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 0.5,
            "height": 40,
            "marginBottom": 20,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Name"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 0.5,
            "height": 40,
            "marginBottom": 20,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 0.5,
            "height": 40,
            "marginBottom": 20,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text
        style={null}
      >
        Register
      </Text>
    </View>
  </View>,
  " ",
]
`;
